### PR TITLE
dynamic casting support in LaravelConfig

### PR DESCRIPTION
**Pull request by YunusEmreNalbant**
Added dynamic casting support to the LaravelConfig class, enabling it to cast configuration values based on their types.

### CODE REVIEW

File: tests/LaravelConfigTest.php [hunk 1]
```diff
@@ -206,4 +207,78 @@ public function it_returns_nested_config_parameters(): void
...
         $this->assertEquals('baz', $response->last()->name);
     }
+
+    /** @test */
+    public function it_returns_boolean_value_for_boolean_type_config_parameter_if_exists(): void
+    {
+        $config = factory(Config::class)->create([
+            'name' => 'yunus.was.here',
+            'val' => '1',
+            'type' => 'boolean',
```
Comment (tkaratug) [inline tests/LaravelConfigTest.php]: Buraya enum case verebiliriz.

File: tests/LaravelConfigTest.php [hunk 2]
```diff
@@ -206,4 +207,78 @@ public function it_returns_nested_config_parameters(): void
...
+        $this->assertTrue($response);
+    }
+
+    /** @test */
+    public function it_returns_integer_value_for_integer_type_config_parameter_if_exists(): void
+    {
+        $config = factory(Config::class)->create([
+            'name' => 'yunus.was.here',
+            'val' => '123456',
+            'type' => 'integer',
```
Comment (tkaratug) [inline tests/LaravelConfigTest.php]: Buraya enum case verebiliriz.

File: tests/LaravelConfigTest.php [hunk 3]
```diff
@@ -206,4 +207,78 @@ public function it_returns_nested_config_parameters(): void
...
+        $this->assertIsInt($response);
+    }
+
+    /** @test */
+    public function it_returns_datetime_value_for_datetime_type_config_parameter_if_exists(): void
+    {
+        $config = factory(Config::class)->create([
+            'name' => 'yunus.was.here',
+            'val' => '2024-02-29 12:00',
+            'type' => 'datetime',
```
Comment (tkaratug) [inline tests/LaravelConfigTest.php]: Buraya enum case verebiliriz.

File: tests/LaravelConfigTest.php [hunk 4]
```diff
@@ -206,4 +207,78 @@ public function it_returns_nested_config_parameters(): void
...
+        $this->assertInstanceOf(Carbon::class, $response);
+    }
+
+    /** @test */
+    public function it_returns_date_value_for_date_type_config_parameter_if_exists(): void
+    {
+        $config = factory(Config::class)->create([
+            'name' => 'yunus.was.here',
+            'val' => '2024-02-29',
+            'type' => 'date',
```
Comment (tkaratug) [inline tests/LaravelConfigTest.php]: Buraya enum case verebiliriz.

File: tests/LaravelConfigTest.php [hunk 5]
```diff
@@ -206,4 +207,78 @@ public function it_returns_nested_config_parameters(): void
...
+        $this->assertInstanceOf(Carbon::class, $response);
+    }
+
+    /** @test */
+    public function it_returns_json_value_for_json_type_config_parameter_if_exists(): void
+    {
+        $config = factory(Config::class)->create([
+            'name' => 'yunus.was.here',
+            'val' => '{"9":[7,8,9],"2":[7,8,9],"31":[10,11,12]}',
+            'type' => 'json',
```
Comment (tkaratug) [inline tests/LaravelConfigTest.php]: Buraya enum case verebiliriz.

**Review by tkaratug — CHANGES_REQUESTED** [review]

File: tests/LaravelConfigTest.php (hunk 1 above)
Comment (YunusEmreNalbant) [reply tests/LaravelConfigTest.php]: güncelledim: https://github.com/tarfin-labs/laravel-config/pull/31/commits/09968a241430042e1093144308c3131bbc9a8e33

**Review by tkaratug — APPROVED** [review]

**Pull request merged**